src/hooks/useCurrentLanguage.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+
+import { TranslatableTextKeys } from "./../models/index";
+
+interface CurrentLanguage {
+  language: TranslatableTextKeys;
+}
+
+export const useCurrentLanguage = (
+  defaultLanguage: TranslatableTextKeys = "en"
+): CurrentLanguage => {
+  const [language, setLanguage] =
+    useState<TranslatableTextKeys>(defaultLanguage);
+  const { i18n } = useTranslation();
+
+  useEffect(() => {
+    setLanguage(i18n.language.slice(0, 2) as never);
+  }, [i18n.language]);
+
+  return { language };
+};
```

Comment:
Why not just

```suggestion
export const useCurrentLanguage = (): TranslatableTextKeys => {
  const { i18n } = useTranslation();
  const language = useMemo(() => i18n.language.slice(0, 2) as TranslatableTextKeys, [i18n.language])
  return language;
};
```
I removed the `defaultLanguage`, as it didn't really do anything valuable before, because of following line resulting in a exception anyways.
https://github.com/LeoTuet/ui-we-want-to-sleep/blob/c72de5a14f3a848eac51d533b542de07025e5414/src/hooks/useCurrentLanguage.ts#L18

Note: NOT TESTED